nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -293,11 +289,39 @@ def _stop(self):
     def _process_config(self, config):
         pass
 
-    def _process_result(self, result):
+    @abc.abstractmethod
+    def _run_iteration(self):
+        """Run one iteration of the experiment"""
+        raise NotImplementedError
+
+    def _process_result(self, result, pre_experiment_result=None):
         pass
 
 
-class SigOptImagenetTrainable(ImagenetTrainable):
+class SupervisedTrainable(BaseTrainable):
+    """
+    Trainable class used to train supervised machine learning experiments
+    with ray.
+    """
+
+    def _run_iteration(self):
+        """Run one epoch of training on each process."""
```

Comment:
Is the word epoch here still applicable?

Reply:
It is because it's in the SupervisedTrainable.